### PR TITLE
add version mapping for the stable ubuntu ppa

### PR DESCRIPTION
**Pull request by wsnipex**
enable uploading to the ubuntu stable ppa if the addon branch is Leia or leia

### CODE REVIEW

**Comment by Rechi** [comment]
> if the addon branch is Leia or leia

This is not what this PR will do.

**Comment by wsnipex** [comment]
hmm? you mind enlightening me?

**Comment by Rechi** [comment]
`PPA_VERSION_MAP` is used to map `version` argument of the `buildPlugin` function call (which defaults to master) to a PPA.
If you want that add-ons are uploaded to stable PPA you have to use the following ppa map.
```groovy
def PPA_VERSION_MAP = [
	'master': 'stable',
	'Leia': 'stable'
]
```
\
*The add-on branch name is (and always will be) only used for the weekly cron trigger for master branch.*

**Comment by wsnipex** [comment]
well, this is not what I had in mind when I added this. The plan always was to upload to stable if the addon branch is leia. We certainly do not want to push master to the stable ppa.

**Comment by Rechi** [comment]
You do not care about the branch name in the addon repo, you care about the targeted Kodi version.
At the time you added it it was correct as stable PPA contained v17.

**Comment by wsnipex** [comment]
What happens when we bump master to v19 (and break abi compatibility), so addon master won't be compatible. At this point there will have to be leia branches in addons.
How do you suggest to handle this, if not via branch name?

**Comment by Rechi** [comment]
As said via the `version` argument for `buildPlugin` function call, which defaults to master if it is not specified.

File: vars/buildPlugin.groovy [hunk 1]
```diff
@@ -42,7 +42,8 @@ def call(Map addonParams = [:])
 	]
 	def PPA_VERSION_MAP = [
 		'master': 'nightly',
-		'Leia': 'unstable'
+		'Leia': 'stable',
+		'leia': 'stable'
```
Comment (Rechi) [inline vars/buildPlugin.groovy]: Version can never be lower case `leia`.
```suggestion
```

**Comment by Rechi** [comment]
@wsnipex stable PPA doesn't have a Kodi version for Ubuntu Disco. Will you upload 18.1 to it, should I wait with merging this PR till 18.2 or just ignore Disco for now?

**Pull request merged**